src/test/java/org/cactoos/io/CloseShieldInputStreamTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2020 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+import org.cactoos.text.TextOf;
+import org.hamcrest.core.IsEqual;
+import org.junit.jupiter.api.Test;
+import org.llorllale.cactoos.matchers.Assertion;
+import org.llorllale.cactoos.matchers.IsText;
+import org.llorllale.cactoos.matchers.Verifies;
+
+/**
+ * Test case for {@link CloseShieldInputStream}.
+ * @since 1.0.0
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+final class CloseShieldInputStreamTest {
+
+    @Test
+    @SuppressWarnings("try")
+    void preventsOriginalStreamToBeClosed() throws Exception {
+        try (FakeInputStream origin = new FakeInputStream()) {
```

Comment:
@baudoliver7 it should be `InputStream` on the left side

Reply:
@andreoss I'm afraid not because we need access to method #isClosed as below :
```java
try (FakeInputStream origin = new FakeInputStream()) {
    // @checkstyle EmptyBlockCheck (2 lines)
    try (InputStream stream = new CloseShieldInputStream(origin)) {
    }
    new Assertion<>(
        "Must not close origin stream",
        origin.isClosed(),
        new IsEqual<>(false)
    ).affirm();
}
```

Reply:
@baudoliver7 I see now, sorry